cmake_modules/CheckAtomicIntrinsic.cmake

```diff
@@ -53,7 +53,9 @@ if( SUPPORT_C11 AND PARSEC_ATOMIC_USE_C11_ATOMICS )
     #include <stdint.h>
     #include <stdatomic.h>
     int main(void) {
-        int32_t where = 0;
+        int32_t where = 0, expected = 0;
+        if( !atomic_compare_exchange_strong( (_Atomic __int32_t*)&where, &expected, 1 ) )
+            return -1;
```

Comment:
The original code returns 0 or 1. This patch makes it return 0 (success), 1 or -1. I guess 1 and -1 can be conflated to failure in both cases. POSIX specifies only that the 8 lower bits are meaningful (https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/functions/V2_chap02.html#tag_15_13), so -1 is not clear.

May I suggest we use EXIT_SUCCESS and EXIT_FAILURE for all return/exit values, since those are POSIX compliant?

Reply:
I thought about harmonizing them but then I realized that it might be helpful to have different return values for the two error cases. That way we can distinguish (in the CMake log) whether the operation is not lock-free or whether we failed to execute it in the first place...

Reply:
return 1 and 2 then, that's probably more clear than 1 and 254